src/secrets/CommandMkdir.ts

```diff
@@ -50,16 +54,48 @@ class CommandMkdir extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsMkdir({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              dirName: secretPath[1],
-              recursive: options.recursive,
-            }),
-          meta,
+        const response = await binUtils.retryAuthentication(async (auth) => {
+          const response =
+            await pkClient.rpcClient.methods.vaultsSecretsMkdir();
+          const writer = response.writable.getWriter();
+          let first = true;
+          for (const [vault, path] of secretPaths) {
+            await writer.write({
+              nameOrId: vault,
+              dirName: path,
+              metadata: first
+                ? { ...auth, options: { recursive: options.recursive } }
+                : undefined,
+            });
+          }
+          await writer.close();
+          return response;
+        }, meta);
+
+        // As git does not track empty directories in version control, we need
+        // to inform the user about this.
+        process.stderr.write(
+          'WARNING: Empty directories will not be affected by modifications to the vault state (version change, sharing/cloning, etc.)',
         );
```

Comment:
I'm not sure how to word the warning for this. This also needs a standard. However, this is the one (and possibly only) situation where this might come up, so we might not need to worry about this.

Also, how do we deal with formatting long strings? The formatter enforces max character length to approximately 80 characters, but it is not easily possible to do that with strings. However, for strings of this length, there are ways of wrapping them elegantly. Is it worth doing that to make the code look cleaner?

```ts
const message = [
  'this is a super long line of',
  'a long string which will all',
  'be printed on the same line.'
].join(' ');

// message = 'this is a super long line of a long string which will all be printed on the same line';
```

Reply:
Don't write a warning for this to `stderr`. This information should be part of documentation and the command description instead.

Reply:
https://github.com/MatrixAI/Polykey-CLI/pull/300#discussion_r1798761117 now that we are no longer sending users a message on stderr, this is no longer relevant